target.py

```diff
@@ -30,10 +30,20 @@ def __init__(self):
         self.points = target_attr.points
         self.rect = self.image.get_rect()
         #location
-        self.rect.x = random.randint(0, DISPLAY_SIZE[0] - target_attr.width)
-        self.rect.y = random.randint(0, DISPLAY_SIZE[1] - target_attr.height)
+        self.rect.x = random.randint(0, DISPLAY_SIZE[0] - target_attr.width - 20) #-20 for edge buffer
+        self.rect.y = random.randint(0, DISPLAY_SIZE[1] - target_attr.height - 20) #-20 for edge buffer
 
     def update(self, bullet):
         if (self.rect.collidepoint(bullet.x, bullet.y)):
             self.kill()
             return self.points
+
+    def spawn_new_target(self, targetSprites, player):
+        self = Target()
+        targetSprites.add(self)
+        if ((self.rect.x > player.x - player.width and self.rect.x < player.x + player.width) 
+            or (self.rect.y > player.x - player.height and self.rect.y < player.x + player.height)):
```

Comment:
Can you explain to me why the second logic (after the or) uses player.x when comparing to self.rect.y?  I understand the +/- player.height here but not in using player.x coordinate.

Also, this if statement is a bit long and hard to understand.  I think you can use the colliderect function of the Rect object provided by pygame.  This function returns a boolean.  Maybe you can do something like if (self.rect.colliderect(player.rect)) ....

One more thing, since you're using an if statement, are you assuming that the second spawn will for sure be created off of the player surface?  How are you certain that the back up target spawn will not be created on top of the player as well?